Sources/TensorFlow/LazyTensor/LazyTensorOperation.swift

```diff
@@ -0,0 +1,551 @@
+        case .DoubleValue(let v): return "Double(\(v))"
+        case .StringValue(let v): return "\"\(v)\""
+        case .BoolArray(let values): return arrayAsString("", values)
+        case .IntArray(let values): return arrayAsString("Int", values)
+        case .FloatArray(let values): return arrayAsString("Float", values)
+        case .DoubleArray(let values): return arrayAsString("Double", values)
+        case .StringArray(let values): return arrayAsString("String", values)
+        case .ConstTensor(let v): return "Const(\(v))"
+        case .TensorDataTypeValue(let v): return dataTypeAsString(v)
+        }
+    }
+
+    private func arrayAsString<T>(_ desc: String, _ values: [T]) -> String {
+        let arrayDesc = (values.map { "\($0)" }).joined(separator: ", ")
+        return "\(desc)[\(arrayDesc)]"
+    }
+
+    private func dataTypeAsString(_ dataType: TensorDataType) -> String {
+        switch dataType._cDataType {
+        case TF_FLOAT: return "float"
+        case TF_DOUBLE: return "double"
+        case TF_INT32: return "int32"
+        case TF_UINT8: return "uint8"
+        case TF_INT16: return "int16"
+        case TF_INT8: return "int8"
+        case TF_STRING: return "string"
+        case TF_COMPLEX64, TF_COMPLEX: return "complex"
+        case TF_INT64: return "int64"
+        case TF_BOOL: return "bool"
+        case TF_QINT8: return "qint8"
+        case TF_QUINT8: return "quint8"
+        case TF_QINT32: return "qint32"
+        case TF_BFLOAT16: return "bfloat16"
+        case TF_QINT16: return "qint16"
+        case TF_QUINT16: return "quint16"
+        case TF_UINT16: return "uint16"
+        case TF_COMPLEX128: return "complex128"
+        case TF_HALF: return "half"
+        case TF_RESOURCE: return "resource"
+        case TF_VARIANT: return "variant"
+        case TF_UINT32: return "uint32"
+        case TF_UINT64: return "uint64"
+        default: assert(false, "Unhandled type: \(dataType._cDataType)")
+        }
+    }
+}
+
+extension LazyTensor: CustomStringConvertible {
+    public var description: String {
+        switch self.handle {
+        case LazyTensor.Handle.concrete(_, let isMaterialized):
+            // TODO: Print the actual concrete value.
+            return isMaterialized ? "conc*" : "conc"
+        case LazyTensor.Handle.symbolic(let op, let index, let isLive):
+            return isLive
+                ? "\(op.nameWithID):\(index)*"
+                : "\(op.nameWithID):\(index)"
+        }
+    }
+}
+
+extension LazyTensorOperation : CustomStringConvertible {
+    public var description: String {
+        let attrsDesc = attrs.map { (name, value) in "\(name): \(value)" }
+        let inputsDesc = inputs.map { (input: Input) -> String in
+            switch input {
+            case Input.single(let lazyTensor):
+                return "\(lazyTensor)"
+            case Input.list(let lazyTensorList): do {
+                    let lazyTensors = lazyTensorList.map{ "\($0)" }
```

Comment:
The indentation here seems incorrect. Also there should be a space before a trailing closure.
```suggestion
                let lazyTensors = lazyTensorList.map { "\($0)" }
```